draft/0005-impersonation.md

```diff
@@ -0,0 +1,279 @@
+```
+
+Logto validates the `userId` and permissions to issue a subject token. If the request is valid, Logto generates a new subject token that represents the identity of the user and returns it in the response. Here is an example of the response:
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "subjectToken":"accVkjcJyb4BWCxGsndESCJQbdFMogUC5PbRDqceLTC",
+  "expiresIn":600
+}
+```
+
+## Additional claims in the access token
+
+### `act`
+
+When `actor_token` presents in the request of token exchange, the access token issued by Logto will contain an additional `act` claim that represents the identity of the impersonated user. The `act` claim is an object that contains the following fields.
+
+1. `sub`: REQUIRED. The identity of the acting user. Typically, this will be the party that is authorized to use the requested security token and act on behalf of the subject.
+
+This is an example JWT access token with the `act` claim.
+
+```json
+{
+  "aud": "https://resource.example.com",
+  "iss": "https://example.logto.app",
+  "exp": 1443904177,
+  "sub": "user2",
+  "act": {
+    "sub": "user1"
+  }
+}
+```
+
+The `act` claim is used to indicate that user1 is acting on behalf of user2.
+
+### `client_id`
+
+A JWT access token in Logto contains a `client_id` claim that represents the client application that requested the token. When a subject token is issued, the `client_id` claim in the subject token is set to the `client_id` of the machine-to-machine application that requested the subject token.
+
+## Example
+
+The following example demonstrates how a support engineer can use the user impersonation feature to troubleshoot an issue on behalf of a user. Let's say the support engineer received a report from a user that they are unable to access a specific resource. The support engineer needs to investigate the issue and verify the user's permissions.
+
+The following sequence diagram illustrates the flow of the user impersonation feature in Logto.
+
+```mermaid
+sequenceDiagram
+    participant SE as Support Engineer
+    participant RO as Resource Owner API
+    participant M2M as Logto Management API
+    participant TS as Tenant's Token Endpoint
+    participant RS as Resource
+    
+    SE ->> RO: Request for subject token
+    RO ->> RO: validate user permission
+    RO ->> M2M: Request for subject token
+    M2M ->> RO: subject token
+    RO ->> SE: subject token
+    
+    SE ->> TS: POST /oidc/token to grant
+    TS ->> SE: access_token (along with other params)
+    
+    SE ->> RS: Fetch user resource with the access token
+    RS ->> SE: Return data
+```
+
+### Step 1: Issue a subject token
+
+The support engineer calls the resource owner's API to issue a subject token for the user who reported the issue. The following request is a simple example of the resource owner's API, the resource owner MUST implement the necessary security controls to validate the support engineer's request according to their business requirements.
```

Comment:
better to add something like "Assume the API is `POST /request-for-subject-token`, which requires ..."